### PR TITLE
feat: new scan workflow

### PR DESCRIPTION
**Pull request by jleach**
# Summary of Changes

We got some feedback letting is know the existing workflow of automatically waiting for, and processing offers and proofs was not awesome. Based on the feedback, we don't automatically process these unless its a proof request packaged with an OOB connection. Otherwise you'll be navigated to the contact details of the connection that was established.

We'll re-visit this when goal codes are implemented and use them to determine what logic to execute. 

I wasn't sure how much dead code to cleanup since some or all of it will be used again in the figure. My initial though was to remove it all - and I'm happy do do so based on feedback.

# Related Issues

bcgov/bc-wallet-mobile#1141

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

**Comment by jleach** [comment]
@bryce-mcmath @wadeking98 Thoughts on what to do with the dead code? My though is it will be used again in the future but I'd prefer to delete it because I have no idea when it will be implemented as part of the goal codes ticket.

File: packages/legacy/core/App/screens/Connection.tsx [hunk 1]
```diff
@@ -72,102 +69,98 @@ const Connection: React.FC<ConnectionProps> = ({ navigation, route }) => {
...
-      }
+      // No goal code, we don't know what to expect next,
+      // navigate to the contact details.
+      navigation
+        .getParent()
+        ?.navigate(TabStacks.HomeStack, { screen: Screens.ContactDetails, params: { connectionId } })
+    }
+
+    return () => {
+      connection === undefined
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/screens/Connection.tsx]: Just curious, what is the purpose of returning a boolean from the cleanup function?

File: packages/legacy/core/App/screens/Connection.tsx [hunk 2]
```diff
@@ -72,102 +69,98 @@ const Connection: React.FC<ConnectionProps> = ({ navigation, route }) => {
...
+          navigation.navigate(Screens.ProofRequest, { proofId: notification.id })
+          dispatch({ isVisible: false })
           break
         }
       }
     }
-  }, [notifications, state])
+
+    return () => {
+      notifications === undefined
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/screens/Connection.tsx]: Same as above

**Review by bryce-mcmath — COMMENTED** [review]
Just a couple comments out of curiosity

File: packages/legacy/core/App/screens/Connection.tsx (hunk 1 above)
Comment (jleach) [reply packages/legacy/core/App/screens/Connection.tsx]: I'm testing to see if it clears up a memory leak waring from metro. My though was the connection or notification below change after the screen is dismissed triggering a useEffect and causing the warning. Might be a long shot but I'm not seeing it ATM.

File: packages/legacy/core/App/screens/Connection.tsx (hunk 2 above)
Comment (jleach) [reply packages/legacy/core/App/screens/Connection.tsx]: See above.

File: packages/legacy/core/App/screens/Connection.tsx (hunk 1 above)
Comment (bryce-mcmath) [reply packages/legacy/core/App/screens/Connection.tsx]: But isn't that a equality operator and not an assignment? I think those two vars are declared with `const` too. But if it's still working somehow we can keep it, that works for me

Comment (jleach) [reply packages/legacy/core/App/screens/Connection.tsx]: Oh, yes, good catch.

**Review by bryce-mcmath — COMMENTED** [review]
Are we still changing it to navigate to the chat screen rather than the contact details screen?

File: packages/legacy/core/App/screens/Connection.tsx (hunk 1 above)
Comment (jleach) [reply packages/legacy/core/App/screens/Connection.tsx]: Removed. Wasn't what I was after.

File: packages/legacy/core/App/screens/Connection.tsx (hunk 2 above)
Comment (jleach) [reply packages/legacy/core/App/screens/Connection.tsx]: Removed. Wasn't what I was after.

**Comment by jleach** [comment]
> Are we still changing it to navigate to the chat screen rather than the contact details screen?

Yes, this navigates to the chat screen now.

File: packages/legacy/core/App/screens/Connection.tsx [hunk 3]
```diff
@@ -72,102 +69,98 @@ const Connection: React.FC<ConnectionProps> = ({ navigation, route }) => {
...
-          navigation.navigate(Screens.CredentialOffer, { credentialId: notificationRecord.id })
-          break
-        case 'ProofRecord':
-          navigation.navigate(Screens.ProofRequest, { proofId: notificationRecord.id })
-          break
-        default:
-          throw new Error('Unhandled notification type')
-      }
+      // No goal code, we don't know what to expect next,
+      // navigate to the contact details.
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/screens/Connection.tsx]: It really looks like it still navigates to the contact details screen

Comment (jleach) [reply packages/legacy/core/App/screens/Connection.tsx]: Yes, didn't push my change + I just updated the docs. Should be fixed now.

**Review by bryce-mcmath — APPROVED** [review]

**Pull request merged**